src/Database/Schema/Index.php

```diff
@@ -0,0 +1,297 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @copyright     Copyright (c) Cake Software Foundation, Inc.
+ *                (https://github.com/cakephp/migrations/tree/master/LICENSE.txt)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Schema;
+
+use RuntimeException;
+
+/**
+ * Index value object
+ *
+ * Models a database index and its attributes.
+ */
+class Index
+{
+    /**
+     * @var string
+     */
+    public const INDEX = TableSchema::INDEX_INDEX;
+
+    /**
+     * @var string
+     */
+    public const FULLTEXT = TableSchema::INDEX_FULLTEXT;
+
+    /**
+     * @var ?array<string>
+     */
+    protected ?array $columns = null;
+
+    /**
+     * @var string
+     */
+    protected string $type = self::INDEX;
+
+    /**
+     * @var string|null
+     */
+    protected ?string $name = null;
+
+    /**
+     * @var array|int|null
+     */
+    protected int|array|null $length = null;
+
+    /**
+     * @var ?array<string>
+     */
+    protected ?array $order = null;
+
+    /**
+     * @var ?array<string>
+     */
+    protected ?array $includedColumns = null;
+
+    /**
+     * @var bool
+     */
+    protected bool $concurrent = false;
+
+    /**
+     * @var string|null The where clause for partial indexes.
+     */
+    protected ?string $where = null;
+
+    /**
+     * Sets the index columns.
+     *
+     * @param array<string>|string $columns Columns
+     * @return $this
+     */
+    public function setColumns(string|array $columns)
```

Comment:
Think we require array|string

Reply:
I ran composer cs-fix and no changes were made 🤷